.circleci/config.yml

```diff
@@ -15,152 +42,362 @@ references:
+            packaging_type=$(mvn help:evaluate -Dexpression=project.packaging -q -DforceStdout | xargs)
+            echo "Packing type: ${packaging_type}"
+            if [[ ${packaging_type} == "bundle" ]]; then
+              group_id=$(mvn help:evaluate -Dexpression=project.groupId -q -DforceStdout | xargs)
+              parent_group_id=$(mvn help:evaluate -Dexpression=project.parent.groupId -q -DforceStdout | xargs)
+              project_name=$(mvn help:evaluate -Dexpression=project.name -q -DforceStdout | xargs)
+              if [[ ${group_id} != "org.jahia.modules" && ${parent_group_id} != "org.jahia.modules" ]]; then
+                echo "Project is not a Jahia modules"
+                circleci-agent step halt
+              else
+                KEYMAKER_CLI_OUTPUT=`java -jar ${KEYMAKER_CLI_JAR} pom -f pom.xml -s`
+                if [[ $(echo $KEYMAKER_CLI_OUTPUT | grep -w "Signature valid" | wc -l) == "1" ]]; then
+                  echo "Signature is already up-to-date"
+                  circleci-agent step halt
+                else
+                  echo "New signature: ${KEYMAKER_CLI_OUTPUT}"
+                  sed -i -e "s%<\(.*\)Jahia-Signature>.*</%<\1Jahia-Signature>${KEYMAKER_CLI_OUTPUT}</%" pom.xml
+                  sed -i -e "s%<\(.*\)jahia-module-signature>.*</%<\1jahia-module-signature>${KEYMAKER_CLI_OUTPUT}</%" pom.xml
+                  git add pom.xml && git commit -m "[ci skip] Updated signature for ${project_name}"
+                  git push -u origin ${CIRCLE_BRANCH}
+                fi
+              fi
+            else
+              echo "There are submodule(s) in this projects"
+            fi
 
 workflows:
   version: 2
-  # This one is run only when triggered with the API, using a GitHub action
-  pull_request:
-    when: << pipeline.parameters.run_pull_request >>
+  # On code change is executed each time new code is pushed to a branch
+  # Current project configuration in circleci only builds on PR + main, so effectively this is only being executed
+  on-code-change:
+    unless: << pipeline.parameters.IS_RELEASE >>
     jobs:
-      - checkout:
-          name: PR Checkout
+      - initialize
       - build:
-          name: PR Build
-          is_pull_request: true
           context: QA_ENVIRONMENT
           requires:
-            - PR Checkout
+            - initialize
       - sonar:
-          name: PR Sonar analysis
-          is_pull_request: true
           context: QA_ENVIRONMENT
           requires:
-            - PR Build
-  on-code-change:
-    jobs:
-      - checkout:
+            - build
+      - publish:
+          context: QA_ENVIRONMENT
+          requires:
+            - build
           filters:
             branches:
               only:
-                - master
+                - << pipeline.parameters.PRIMARY_RELEASE_BRANCH >>
                 - /feature-.*/
                 - /[0-9]*_x/
                 - /[0-9]*_[0-9]*_x/
                 - /[0-9]*-[0-9]*-X-BRANCH/
```

Comment:
Do we still need the branch names ?

Reply:
I removed it, we will add it on the branches if necassary